### PR TITLE
58 display createpost on the profile page

### PR DESCRIPTION
**Pull request by Matt-Tan15**
Updated the sharePost component can now create posts and added the component to profile page

### CODE REVIEW

File: src/shared/components/SharePost.tsx [hunk 1]
```diff
@@ -8,27 +8,38 @@ import {
...
+  const { form, handleChange, resetForm } = useForm<BaseCreatePostmodel>({
     media: "",
     content: "",
     categoryName: "",
   });
-  const { media, content, categoryName } = form;
+
+  const { content, categoryName } = form;
+  const { onSelectFile, preview, onRemove, selectedFile } = useImageInput();
+  const endPoint = process.env.REACT_APP_BASE_URL + "/posts";
```
Comment (raylu511) [inline src/shared/components/SharePost.tsx:32]: I would move this inside of handleSubmit.

File: src/shared/components/SharePost.tsx [hunk 2]
```diff
@@ -38,11 +49,10 @@ const SharePost = ({ profilePic, sx, onClick, isPost }: SharePostProps) => {
         display: "flex",
         flexDirection: "column",
         justifyContent: "space-between",
-        backgroundColor: "#ffffff",
-        maxWidth: "600px",
-        margin: "auto",
+        backgroundColor: theme.backgroundColor,
```
Comment (raylu511) [inline src/shared/components/SharePost.tsx:52]: nit: I would destructure these properties from the theme

File: src/shared/components/SharePost.tsx [hunk 3]
```diff
@@ -57,19 +67,38 @@ const SharePost = ({ profilePic, sx, onClick, isPost }: SharePostProps) => {
...
+              name="content"
+              value={content}
+              onChange={handleChange}
+              sx={{
+                color: "black",
+                backgroundColor: "#ededed",
+                paddingLeft: "30px",
+                borderRadius: "30px",
+              }}
+              placeholder={"What's on your mind..."}
```
Comment (raylu511) [inline src/shared/components/SharePost.tsx:82]: remove curly braces if you're rendering a string

File: src/shared/components/SharePost.tsx [hunk 4]
```diff
@@ -83,59 +112,49 @@ const SharePost = ({ profilePic, sx, onClick, isPost }: SharePostProps) => {
...
                   alignItems: "center",
                   pr: "15px",
                   ":hover": {
                     cursor: "pointer",
-                    backgroundColor: "#ededed",
+                    backgroundColor: theme.navButtons,
                   },
                 }}
               >
+                {" "}
```
Comment (raylu511) [inline src/shared/components/SharePost.tsx:129]: Remove

File: src/shared/components/SharePost.tsx (hunk 1 above)
Comment (Matt-Tan15) [reply src/shared/components/SharePost.tsx:32]: gotcha

File: src/shared/components/SharePost.tsx (hunk 2 above)
Comment (Matt-Tan15) [reply src/shared/components/SharePost.tsx:52]: sounds good

File: src/shared/components/SharePost.tsx (hunk 4 above)
Comment (Matt-Tan15) [reply src/shared/components/SharePost.tsx:129]: done

File: src/shared/components/SharePost.tsx (hunk 3 above)
Comment (Matt-Tan15) [reply src/shared/components/SharePost.tsx:82]: gotcha

**Review by raylu511 — APPROVED** [review]

**Review by DLiSWE — APPROVED** [review]

**Pull request merged**